app/services/publish_links.rb

```diff
@@ -27,7 +31,14 @@ def links_to_update
 
   def updated_links
     links_to_update.merge(existing_links) do |_, new_links, old_links|
-      (old_links || []).concat(new_links).uniq
+      old_links_for_type =
+        if tag_mapping.delete_source_link?
+          old_links - [source_content_id]
+        else
+          old_links
+        end
+
+      (old_links_for_type || []).concat(new_links).uniq
```

Comment:
@jackscotti I take that back, it isn't a bug. The reason `old_links_for_type` can be nil because when we are tagging a given content item for the first time and there are no links to return. The links hash is empty and therefore `old_links` is nil. Does that make sense?